static/skin/viewer.js

```diff
@@ -266,6 +266,8 @@ function onClickEvent(e) {
       if ( viewerSettings.linkBlockingEnabled ) {
         return blockLink(target);
       }
+    } else {
+      target.setAttribute("target", "content_iframe");
```

Comment:
This change is nice.
I even wonder if you would have done this at first instead of introducing complex sandboxing.

Reply:
Well, the sandboxing was necessary to prevent scripts in the ZIM accessing external resources ("phoning home") -- a real issue with Zimit ZIMs -- and also to prevent top-level navigation. We tried lots of combinations...

Reply:
Yeah yeah, but I think is better for us to don't count the time we have spend on this 😉 

---

Maybe (as almost all the time), the best would have been to fix the root cause and fix the generated zims.
After all, the SW embedded in the zim files is able to catch a request to `www.foo.com/bar` and do a request to `/A/www.foo.com/bar` instead. We could (and can) make the SW refuse request to unhanded url.

Reply:
Sandboxing was needed to prevent programmatic escape from the iframe (see kiwix/kiwix-tools#604). Unfortunately this trick [doesn't work perfectly in Chromium](https://github.com/kiwix/libkiwix/pull/944#issuecomment-1533056709)